src/cache/http_cache_provider.cpp

```diff
@@ -0,0 +1,218 @@
+    return false;
+  }
+
+  // The default port for HTTP is 80 (other services, such as MinIO, may use other ports).
+  if (m_port < 0) {
+    m_port = 80;
+  }
+
+  // If the path is non-empty, it needs to start with a slash.
+  if ((!m_path.empty()) && m_path[0] != '/') {
+    m_path = "/" + m_path;
+  }
+
+  m_ready_for_action = true;
+  return true;
+}
+
+bool http_cache_provider_t::is_connected() const {
+  return m_ready_for_action;
+}
+
+void http_cache_provider_t::disconnect() {
+  m_ready_for_action = false;
+}
+
+std::string http_cache_provider_t::get_object_url(const std::string& key) {
+  std::ostringstream ss;
+  ss << "http://" << m_host << ":" << m_port << m_path << "/" << key;
+  return ss.str();
+}
+
+cache_entry_t http_cache_provider_t::lookup(const hasher_t::hash_t& hash) {
+  const auto key = remote_key_name(hash.as_string(), CACHE_ENTRY_FILE_NAME);
+  try {
+    // Try to get the cache entry item from the remote cache.
+    return cache_entry_t::deserialize(get_data(key));
+  } catch (const std::exception& e) {
+    // We most likely had a cache miss.
+    debug::log(debug::log_level_t::DEBUG) << e.what();
+    return cache_entry_t();
+  }
+}
+
+void http_cache_provider_t::add(const hasher_t::hash_t& hash,
+                                const cache_entry_t& entry,
+                                const std::map<std::string, expected_file_t>& expected_files) {
+  const auto hash_str = hash.as_string();
+
+  // Upload (and optinally compress) the files to the remote cache.
+  for (const auto& file_id : entry.file_ids()) {
+    const auto& source_path = expected_files.at(file_id).path();
+
+    // Read the data from the source file.
+    auto data = file::read(source_path);
+
+    // Compress?
+    if (entry.compression_mode() == cache_entry_t::comp_mode_t::ALL) {
+      debug::log(debug::DEBUG) << "Compressing " << source_path << "...";
+      data = comp::compress(data);
+    }
+
+    // Upload the data.
+    const auto key = remote_key_name(hash_str, file_id);
+    set_data(key, data);
+  }
+
+  // Create a cache entry file.
+  const auto key = remote_key_name(hash_str, CACHE_ENTRY_FILE_NAME);
+  set_data(key, entry.serialize());
+}
+
+void http_cache_provider_t::get_file(const hasher_t::hash_t& hash,
+                                     const std::string& source_id,
+                                     const std::string& target_path,
+                                     const bool is_compressed) {
+  const auto key = remote_key_name(hash.as_string(), source_id);
+  auto data = get_data(key);
+  if (is_compressed) {
+    data = comp::decompress(data);
+  }
+  file::write(data, target_path);
+}
+
+std::string http_cache_provider_t::get_path() const {
+  return m_path;
+}
+
+std::vector<std::string> http_cache_provider_t::get_header(const std::string& method,
+                                                           const std::string& key) const {
+  std::ignore = method;
+  std::ignore = key;
```

Comment:
Right now I prefer the old way of just leaving out the variable names of the function, i.e:

```c++
std::vector<std::string> http_cache_provider_t::get_header(const std::string& /* method */,
                                                           const std::string& /* key*/) const {
```

...and drop the use of `std:ignore`.